core/services/relay/relay.go

```diff
@@ -144,6 +144,10 @@ func (r *relayerAdapter) NewFunctionsProvider(ctx context.Context, rargs types.R
 	return r.Relayer.NewFunctionsProvider(rargs, pargs)
 }
 
+func (r *relayerAdapter) NewPluginProvider(ctx context.Context, rargs types.RelayArgs, pargs types.PluginArgs) (types.PluginProvider, error) {
+	return nil, fmt.Errorf("unexpected call to NewPluginProvider: did you forget to wrap relayerAdapter in a relayerServerAdapter?")
```

Comment:
This only happens if you configure a non-LOOPP relayer without using `NewRelayerServerAdapter`, i.e. if using `NewRelayerAdapter`.

Reply:
+1 to @jmank88 

if it's always wrong to call this func on relayAdapter, then why does relayAdapter need to satisfy the interface?

Reply:
Is there somewhere sooner/higher-up in the stack where we could do an equivalent check, but as a type cast to an interface with this method instead?

Reply:
I don't think so because both the RelayerAdapter and the RelayerServerAdapter are still used.

To clarify the set up looks something like this:

- all relayers currently implement the old types.Relayer interface
- they get adapted to the loop.Relayer via the RelayerAdapter (in all cases, even where we use the RelayerServerAdapter since that wraps the RelayerAdapter)
- then things are forked:
   - for LOOPP relayers, the RelayerAdapter gets wrapped by the RelayerServer, which intercepts calls to NewPluginProvider and dispatches them to the product-specific function.
    - for non-LOOPP relayers, we use RelayerServerAdapter, which implements the same logic but in non-GRPC terms.
    
The only way I can think of to remove the noop'ing `NewPluginProvider` method would be to create a separate type for the relayerAdapter that doesn't include the method. But that would mean the relayer adapter would no longer satisfy the relayer type, and would cause all sorts of type issues in chainlink-relay.